scripts/codex_autoreview.py

```diff
@@ -0,0 +1,409 @@
+            },
+        ),
+    )
+
+
+def _post_inline_comment(
+    *,
+    owner: str,
+    repo: str,
+    pr_number: int,
+    token: str,
+    commit_id: str,
+    path: str,
+    line: int,
+    body: str,
+) -> dict[str, Any]:
+    payload = {
+        "body": body,
+        "commit_id": commit_id,
+        "path": path,
+        "side": "RIGHT",
+        "line": line,
+    }
+    return cast(
+        dict[str, Any],
+        _github_api(
+            "POST",
+            f"https://api.github.com/repos/{owner}/{repo}/pulls/{pr_number}/comments",
+            token=token,
+            data=payload,
+        ),
+    )
+
+
+def main() -> None:
+    repo_full = os.environ.get("GITHUB_REPOSITORY", "")
+    if not repo_full or "/" not in repo_full:
+        print("GITHUB_REPOSITORY missing or invalid", file=sys.stderr)
+        sys.exit(1)
+    owner, repo = repo_full.split("/", 1)
+    token = os.environ.get("GITHUB_TOKEN") or os.environ.get("GH_TOKEN")
+    if not token:
+        print("GITHUB_TOKEN not provided", file=sys.stderr)
+        sys.exit(1)
+
+    event = _get_event()
+    if "pull_request" not in event:
+        print("This workflow must be triggered by a pull_request event", file=sys.stderr)
+        sys.exit(1)
+    pr = event["pull_request"]
+    pr_number = int(pr["number"]) if "number" in pr else int(event.get("number", 0))
+    head_sha = pr.get("head", {}).get("sha")
+    if not head_sha:
+        print("Missing head commit SHA in event", file=sys.stderr)
+        sys.exit(1)
+
+    changed = _list_changed_files(owner, repo, pr_number, token)
+
+    # Load guidelines and compose the single large prompt
+    guidelines = _load_guidelines()
+    repo_root = Path(os.environ.get("GITHUB_WORKSPACE", ".")).resolve()
+    prompt = _compose_prompt(guidelines, repo_root, changed, pr)
+
+    print("Running Codex to generate review findings...", flush=True)
+    try:
+        output = _run_codex(prompt)
+    except Exception as e:  # pragma: no cover - robust reporting
+        print(f"Codex execution failed: {e}", file=sys.stderr)
+        sys.exit(2)
+
+    # Parse JSON output
+    try:
+        result = json.loads(output)
+    except json.JSONDecodeError as e:
+        print("Model did not return valid JSON:")
+        print(output)
+        print(f"JSON error: {e}", file=sys.stderr)
+        sys.exit(3)
+
+    findings: list[dict[str, Any]] = list(result.get("findings", []) or [])
+    overall = str(result.get("overall_correctness", "")).strip() or "patch is correct"
+    overall_explanation = str(result.get("overall_explanation", "")).strip()
+    overall_conf = result.get("overall_confidence_score")
+
+    summary_lines = [
+        "Codex Autonomous Review:",
+        f"- Overall: {overall}",
+        f"- Findings: {len(findings)}",
+    ]
+    if overall_explanation:
+        summary_lines.append("")
+        summary_lines.append(overall_explanation)
+    if isinstance(overall_conf, int | float):
```

Comment:
[P0] Use of union in isinstance raises at runtime

In scripts/codex_autoreview.py, `isinstance(overall_conf, int | float)` will raise `TypeError: isinstance() arg 2 must be a type, a tuple of types, or a union` at runtime because `int | float` creates a `types.UnionType` object that `isinstance` does not accept as its second argument. This will trigger on every run when building the review summary, causing the workflow to fail regardless of the model output. Replace the check with a tuple of types.
```suggestion
    if isinstance(overall_conf, (int, float)):
```